config/config_test.go

```diff
@@ -63,14 +65,31 @@ func TestInvalidConfigs(t *testing.T) {
 	for k, v := range invalidConfigs {
 		t.Run(k, func(t *testing.T) {
 			filename := mkConfig(t, v)
-			_, err := ReadConfigFile(filename, "", "")
+			_, err := ReadConfigFile(filename, "", "", "")
 			if err == nil {
 				t.Fatal("this config must generate an error")
 			}
 		})
 	}
 }
 
+func TestConfigDir(t *testing.T) {
+	colorGreen := "\033[32m"
+	colorReset := "\033[0m"
+
+	cwd, _ := os.Getwd()
+	fmt.Println(string(colorGreen), cwd, string(colorReset))
+	for k, v := range validConfigs {
+		t.Run(k, func(t *testing.T) {
+			filename := mkConfig(t, v)
```

Comment:
I feel like this test should specifically test the case where there are several config files in a config.d subdirectory.
mkConfig just creates a single file and returns the filename so this seems similar to the "ValidConfig" test above.

Perhaps we can modify mkConfig to take a list of body strings alongside the body, which if it is not empty will create additional files nested under "config.d" directory.

Then, after reading, check if data from all the files got correctly read and parsed.

Also would be worth adding an assertion on the behavior I was staging above (i.e. that ConfigVersion etc are invariant after the specified config rile) so that no one accidentally breaks that in the future.

Reply:
You could also test the same code paths, but in the case where reading fails (probably less important since such a test already exists)